docs/proposal-csv.md

```diff
@@ -0,0 +1,112 @@
+# Proposal for Recording Points in CSV
+
+## Introduction Details
+
+The goal of this proposal is to define a solution to efficiently record and manage the points of our Discord bot users. The aim is to create a storage system that allows filtering information by user, month, and year, facilitating both manual queries and automated processing.
+
+## Background
+
+Initially, the idea of having a single CSV file with all the records was considered. However, some concerns arose:
+
+- **Ease of use:** A single large file can be difficult to handle and view, especially for non-technical users.
+- **Performance:** Processing a very large file can affect query efficiency.
+- **Scalability:** As the amount of data grows, it will become harder to find and filter the necessary information.
+
+Therefore, it is proposed to separate the information into multiple files, organized into folders.
+
+## Proposed Solution
+
+### File Organization
+
+The idea is to store the records in separate CSV files, organized in a folder structure following the hierarchy below:
+
+```md
+/data
+├── <YEAR>
+│├── \<MONTH\>
+││├── <user_id>.csv
+```
+
+**Example:**
+
+```md
+/data
+├── 2025
+│ ├── 03
+│ │ ├── 1007642713530839213.csv (Points for this user in March)
+│ │ ├── 123456789012345678.csv
+│ │ ├── ...
+│ ├── 04
+│ │ ├── ...
+```
+
+### CSV Format
+
+Each CSV file will contain the following fields, enabling filtering and point calculation:
+
+- `user_id`
+- `user_name`
```

Comment:
Maybe this is not needed to repeat each time, we could get the user name from the user_id on discord, so we can save space and use the user's real-time username.

Reply:
also, maybe the user_id is not needed as we have it on the file name?